.github/workflows/docker-builds.yml

```
@@ -0,0 +1,59 @@
+name: Docker Builds
+
+on:
+  push:
+    branches:
+      - '**'
+    tags:
+      - '*.*.*'
+  pull_request:
+    branches:
+      - 'master'
+
+jobs:
+  setup-docker-builds:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get containers to build
+        id: get_containers
+        run: python3 -c 'import json; import os; os.chdir("containers"); print("::set-output name=containers::%s" % json.dumps([os.path.normpath(dirpath) for dirpath, dirnames, filenames in os.walk(".") if "Dockerfile" in filenames]))'
```

I think a multi-line script would be easier to maintain if we ever need to.
```suggestion
        run: |
           python3 -c '
           import json
           import os
           os.chdir("containers")
           print(
               "::set-output name=containers::%s"
               % json.dumps([
                       os.path.normpath(dirpath)
                       for dirpath, _, filenames in os.walk(".")
                       if "Dockerfile" in filenames
                   ]
           ))
           '
```

---

As an alternative to the ``os.walk``, a ``**``-glob can succinctly find the files. I wouldn't say one is more readable than the other, though.
```suggestion
        run: |
           python3 -c '
           import json
           import pathlib
           os.chdir("containers")
           print(
               "::set-output name=containers::%s"
               % json.dumps([
                       str(path.parent.relative_to("containers"))
                       for path in pathlib.Path("containers").glob("**/Dockerfile")
                   ]
           ))
           '
```